modules/social_features/social_group/modules/social_group_invite/src/Controller/SocialGroupInvitationOperations.php

```diff
@@ -59,8 +59,8 @@ public function accepted(Request $request, GroupContentInterface $group_content)
   /**
    * {@inheritdoc}
    */
-  public function checkAccess(GroupContentInterface $group_content) {
-    $invited = $group_content->get('entity_id')->getString();
+  public function checkAccess(GroupRelationshipInterface $group_content) {
```

Comment:
Looking at the checkAccess in the parent class

```
  /**
   * Checks if this current has access to update invitation.
   *
   * @param \Drupal\group\Entity\GroupRelationshipInterface $group_content
   *   Invitation entity.
   *
   * @return \Drupal\Core\Access\AccessResult
   *   Access check result.
   */
  public function checkAccess(GroupRelationshipInterface $group_content) {
    $invited_user_id = $group_content->getEntityId();
    $group = $group_content->getGroup();

    // Plugin is not installed.
    if (!$group->getGroupType()->hasPlugin('group_invitation')) {
      return AccessResult::forbidden();
    }

    $membership = $this->membershipLoader->load($group, $this->currentUser());
    $current_state = $group_content->get('invitation_status')->value;

    // Only allow user accept/decline own invitations.
    if ($invited_user_id == $this->currentUser()->id() && !$membership && (int) $current_state === GroupInvitation::INVITATION_PENDING) {
      return AccessResult::allowed();
    }

    return AccessResult::forbidden();
  }
```

I'm wondering if we need our own implementation for it, it seems it now covers the scenario you can only accept/decline own invitations. But also comes with additional checks that seems relevant.

Reply:
I guess it covers a case when user is already a member - so, he will still have access to his invites, otherwise he will get access denied

here is a related issue: https://www.drupal.org/project/social/issues/3214990